### PR TITLE
add real progress stats

### LINKED ISSUE #81: add hyperdrive-stats
for progress information, see https://github.com/datproject/dat-desktop/issues/19#issuecomment-255356234

### PR DESCRIPTION
**Pull request by juliangruber**
closes #81 

Add real progress stats to the UI!

@joehand do we want to integrate this into dat-js? Would be easier to consume this way.

### CODE REVIEW

File: lib/live-stream.js [hunk 1]
```diff
@@ -53,6 +55,11 @@ function streamData (args, archives, cb) {
               cb()
             })
           })
+          dat.hyperStats = Stats({
+            archive: dat.archive,
+            db: sub(args.db, `${encoding.encode(dat.key)}-stats`)
```
Comment (joehand) [inline lib/live-stream.js]: Whats the reason for using a sub level here instead of the dat.db?

**Comment by joehand** [comment]
> @joehand do we want to integrate this into dat-js? Would be easier to consume this way.

Yes! I do have block progress in there now, but am planning on moving to using hyperdrive stats as one of the next things I do. 

I need to review hyperdrive stats a bit, last time I looked there was something missing but I think it's all there now. Will go through this today!

File: lib/live-stream.js (hunk 1 above)
Comment (juliangruber) [inline lib/live-stream.js]: so we're sure never to have a key collision

File: lib/live-stream.js [hunk 2]
```diff
@@ -53,6 +55,11 @@ function streamData (args, archives, cb) {
...
+          })
+          dat.hyperStats.on('update', () => cb())
```
Comment (yoshuawuyts) [inline lib/live-stream.js]: Perhaps out of scope for this patch, but this pattern feels like a smell - now that we've got dat-js in here we should think of a way to actually move data around (:

Comment (juliangruber) [inline lib/live-stream.js]: > we should think of a way to actually move data around (:

can you expand on that?

**Comment by juliangruber** [comment]
let's iterate on this in future work

**Comment by yoshuawuyts** [comment]
the cb() statement increments a counter rn to force a refresh instead of
passing new data - feel that's not ideal; let's talk about it next week
when you're around and maybs we can find something that works well (:

**Comment by juliangruber** [comment]
yeah definitely! i was curious about this, why do we not follow the "normal" choo pattern and just pass data around?

**Pull request merged**